### PR TITLE
add quoting to split_part

### PR DESCRIPTION
**Pull request by patkearns10**
This is a:
- [ ] bug fix PR with no breaking changes — please ensure the base branch is `main`
- [x] new functionality — please ensure the base branch is the latest `dev/` branch
- [ ] a breaking change — please ensure the base branch is the latest `dev/` branch

## Description & motivation

https://dbt-labs.slack.com/archives/C010X3HKX2L/p1648149203621259

:grey_question: Hi! Quick question about the split_part macro in dbt-utils:
documentation says this `{{ dbt_utils.split_part(string_text='1,2,3', delimiter_text=',', part_number=1) }}`
but I get an error unless I put the `delimiter_text` inside a double quote and a single quote like this `{{ dbt_utils.split_part(string_text='1,2,3', delimiter_text="','", part_number=1) }}` is this expected?

Made an update to add quoting as arguments. 
I did this in a way that is nonbreaking to current usage, but adds functionality if you want to avoid weird input syntax `"','"`

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [x] Snowflake
- [ ] I followed guidelines to ensure that my changes will work on "non-core" adapters by:
    - [x] dispatching any new macro(s) so non-core adapters can also use them (e.g. [the `star()` source](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/sql/star.sql))
    - [ ] using the `limit_zero()` macro in place of the literal string: `limit 0`
    - [ ] using `dbt_utils.type_*` macros instead of explicit datatypes (e.g. `dbt_utils.type_timestamp()` instead of `TIMESTAMP`
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

**Comment by patkearns10** [comment]
before making changes:

Screen Shot 2022-03-25 at 8 33 17 AM

Add quoting to both input fields:
Screen Shot 2022-03-25 at 9 55 00 AM

add integration tests:
Screen Shot 2022-03-25 at 10 28 48 AM

updated docs:
Screen Shot 2022-03-25 at 11 11 29 AM

**Comment by joellabes** [comment]
@patkearns10 I had a look at the standard set across other macros, and things that need quotes to be passed through (e.g. datediff) all require double quotes. I think this is actually a documentation issue, not something that needs to be fixed in code. 

I'm equally unexcited by adding quoting params in this macro as i was lowercase params in the last one - it means that now there's two ways to do the same thing; how do i know which one to use? What happens if I provide conflicting arguments? 

```sql
{{ dbt_utils.split_part("'fixed_string'", "'_'", quote_string=False) }} -- equivalent, which one is better?
{{ dbt_utils.split_part('fixed_string', "'_'", quote_string=True) }} -- equivalent, which one is better?
```

```sql
{{ dbt_utils.split_part("'fixed_string'", "'_'", quote_string=True) }} -- NOW THERE'S TOO MANY QUOTES 😭 
```

As a side note, this would have been another good place to use the table-powered tests as opposed to checking the rendered string; if/when #513 moves forward, we wouldn't want this test to start failing for a non-breaking change. 

I'm happy to either close this or see it repurposed into a documentation PR (or for you to point out the reasons I've got this wrong!) Sorry that the work (and you added tests! on the first go! ⭐ ) don't get to see the light of day - this is another reminder for me to get a move on with the contribution guidelines that ask for issues in advance of PRs 😬

**Comment by patkearns10** [comment]
All set here @joellabes

File: README.md [hunk 1]
```diff
@@ -965,9 +965,15 @@ This macro calculates the difference between two dates.
...
+**Args**:
+- `string_text` (required): Text to be split into parts. 
+- `delimiter_text` (required): Text representing the delimiter to split by.
+- `part_number` (required): Requested part of the split (1-based). If the value is negative, the parts are counted backward from the end of the string.
+
 **Usage:**
 ```
-{{ dbt_utils.split_part(string_text='1,2,3', delimiter_text=',', part_number=1) }}
+{{ dbt_utils.split_part(string_text='some_column', delimiter_text="','", part_number=1) }}
+{{ dbt_utils.split_part(string_text='1,2,3', delimiter_text="'|'", part_number=1) }}
```
Comment (joellabes) [inline README.md:976]: `string_text` needs double quotes and the pipe delimiter, right? 
```suggestion
{{ dbt_utils.split_part(string_text="'1|2|3'", delimiter_text="'|'", part_number=1) }}
```

File: README.md [hunk 2]
```diff
@@ -965,9 +965,15 @@ This macro calculates the difference between two dates.
...
 
+**Args**:
+- `string_text` (required): Text to be split into parts. 
+- `delimiter_text` (required): Text representing the delimiter to split by.
+- `part_number` (required): Requested part of the split (1-based). If the value is negative, the parts are counted backward from the end of the string.
+
 **Usage:**
 ```
-{{ dbt_utils.split_part(string_text='1,2,3', delimiter_text=',', part_number=1) }}
+{{ dbt_utils.split_part(string_text='some_column', delimiter_text="','", part_number=1) }}
```
Comment (joellabes) [inline README.md:975]:
```suggestion
When referencing a column, use one pair of quotes. When referencing a string, use single quotes enclosed in double quotes.
{{ dbt_utils.split_part(string_text='column_to_split', delimiter_text='delimiter_column', part_number=1) }}
```

**Review by joellabes — APPROVED** [review]
⭐  thanks @patkearns10!

**Pull request merged**